### ISSUE TITLE
Suggestions for changes to core design for 1.0 or later.

### ISSUE THREAD

**Issue by TricksterWolf** [blackhole12/PonyClicker#37]
The primary problem with PonyClicker is the interface is terrible.  You can only get a few ponies, when the ponies are best part of PonyClicker.  The more ponies you get, the harder it is to see them.

The spirograph is cute, but unnecessary.  Rather than try to make a cookie-cutter version of Cookie Clicker, I suggest the following format.

The default center screen shows one pony (large) and thumbnailed boxes for each other pony (small).  The boxes are highlighted if that pony is a friend to this pony, and darkened if not.  Clicking on a box changes to that pony.  Right-clicking on a box purchases a friendship to that pony.  The friendships will be remembered rather than reset: your new pony has no friends.  The points you get for clicking on a particular pony will vary based on how many friends that particular pony has, plus other factors.

Now you can have a planned order in which ponies appear, and special achievements for hidden things that should come naturally to the player if they're a brony: get the Mane 6 in a clique, get the CMC in a clique, get the princesses in a clique, get at least 5 villains in a clique, click each of the Mane 6 100 times, etc. etc.

**Comment by blackhole12** [comment]
The current design change to address the pony visibility problem is going to make the wheel grow in size, and then let you spin it around to see different ponies. This works very nicely on a mobile screen.

Long term, a solution like this is preferable, although I think it will still run into display problems when there are 30+ ponies, simply because that's a lot of ponies. However, **this requires a save format change**, and I will recommend it as a change for v1.1 instead. I feel like the display here could use some work - perhaps have one pony be in the center of a circle and have them be selected, and then have a wheel of ponies surrounding them that can be spun so it's still easy to pick ponies on mobile?

**Comment by blackhole12** [comment]
As an aside, the development milestones presented here assume that I have to implement most of the features. If someone else wants to spend the time to make this happen and can successfully merge it without too much trouble, this can be pushed to the v1.0 release.

**Comment by TricksterWolf** [comment]
I'm working with my fiancee on this right now.  It's a pretty comprehensive set of changes (including savefile incompatibility) so maybe 2.0 would be more appropriate.  The largest changes are the selective friendship system, scaling back the speed and advancement of the game dramatically (so it's fun to play for more than a few minutes), putting benchmark locks in place which will unlock by advancing through the episode list, and pushing back the legacy benefits to late game (so you can't get to 1 trillion smiles in a matter of seconds).

**Comment by blackhole12** [comment]
Can the per episode benchmarks be combined with #75? It seems like a similar idea if benchmarks are represented through unique unlockable buildings.

**Comment by TricksterWolf** [comment]
Not quite.  The problem is that the episode requirements need a more detailed representation of the friendship system, e.g., you earn "Look Before You Sleep" when AJ and Rarity are friends.  It needs the "ponies come in order" and "friendships are distinct" at the very least if we want to support per-episode feats.  Otherwise it's too difficult to come up with 100 different things that count for each episode, and I think it would be better if we plan for those to match the advanced system.

**Comment by TricksterWolf** [comment]
I've forked this and played around with it a bit, and I'm not sure how if (time-wise) it would be worth trying to screw with it on my own rather than provide you with a clear list of changes.  The former I can do but it would be much slower b/c my time commitment is limited until after a couple of surgeries I'm getting next month.

I've figured out how to handle the interface and can put together a comprehensive list of episode achievements and how they will work.  If you want I can put together the relevant information in a series of Google docs and link them here.  The interface issue is probably easiest to handle by keeping the pony wheel but having it hold n-1 ponies.  The "active" pony would be placed just above the wheel (or above left) and this would be the only boopable pony.  Clicking on a different pony would swap the active pony, and click-drag or touch-drag from the active pony to a different pony would create a friendship if smile funds exist.  I think it should be scaleable but the problem I'm seeing is drawing the friendships is more complicated; the wheel can be a single svg but the active-pony-to-ponies-in-wheel friendships would need to be drawn dynamically as the wheel spins which seems like that might be too graphic intensive.  Also the individiual friendships would need to be remembered (probably an adjacency list which should be fine b/c the ponies will be purchased in a pre-set order).

**Comment by TricksterWolf** [comment]
Er, actually: an option around the graphics-intensive suggestion would be to indicate ponies who are friends with a background circle or something.  That would allow the wheel to be constructed as a single svg as before, so I think that will work.

**Comment by blackhole12** [comment]
I have a lot of other things that need to be put into v1.1, so I'm not going to take this on just yet. I should mention that you can always cache things on a canvas and then rotate the canvas itself. I don't really consider a huge change like this to be time critical because the core game is at least playable now.

**Comment by TricksterWolf** [comment]
No problem.  Just let me know when you want details and I'll make them
available then.

On Sun, May 31, 2015 at 12:10 PM, Erik McClure <notifications@github.com>
wrote:

> I have a lot of other things that need to be put into v1.1, so I'm not
> going to take this on just yet. I should mention that you can always cache
> things on a canvas and then rotate the canvas itself. I don't really
> consider a huge change like this to be time critical because the core game
> is at least playable now.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/blackhole12/PonyClicker/issues/37#issuecomment-107213366>
> .
>



-- 
Trick Question
